gateway-sp-comms/src/single_sp/update.rs

```diff
@@ -257,57 +258,137 @@ fn read_auxi_check_from_tlvc(data: &[u8]) -> Result<[u8; 32], UpdateError> {
     })
 }
 
+/// Isolate extraction of bootleby from old-format archives.
+/// TODO: Purge old-format archives and delete this code.
+fn bootleby_from_old_style_archive(
+    image: Vec<u8>,
+    log: &Logger,
+) -> Result<Vec<u8>, UpdateError> {
+    // Try the pre-v1.2.0 Bootleby archive format.
+    let cursor = Cursor::new(image.as_slice());
+    let mut archive = zip::ZipArchive::new(cursor).map_err(|zip_error| {
+        // Return the original Hubris Archive error instead
+        // of our attempted zip extraction error.
+        HubtoolsError::ZipError(zip_error)
+    })?;
+
+    let mut rot_image = vec![];
+    for i in 0..archive.len() {
+        let mut file = match archive.by_index(i) {
+            Ok(file) => file,
+            Err(e) => {
+                error!(
+                    log,
+                    "did not find bootleby.bin in old-style archive: {e}"
+                );
+                return Err(UpdateError::InvalidArchive);
```

Comment:
`zip`'s API is a little weird, but I don't think this error is correct, right? E.g., you could have an archive with 10 files, file 7 has an I/O error, and `bootleby.bin` is file 8. We should probably log the error but `continue`, I think.

Reply:
Do we want to proceed using an archive that has a file in it that can't be parsed?